File: src/tmp/Java/src/com/dbc/algorithm/ReferOffer/FeiBoNaQiShuLieLcof.java
package com.dbc.algorithm.ReferOffer;

public class FeiBoNaQiShuLieLcof {
    public int fib(int n) {
        if (n == 0) return 0;
        if (n == 1) return 1;

        int f1 = 0, f2 = 1;
        for (int i = 0; i < n - 1; i++) {
            int temp = (f1 + f2) % 1000000007;
            f1 = f2;
            f2 = temp;
        }
        return f2;
    }
}

File: src/tmp/Python/refer-offer/fei_bo_na_qi_shu_lie_lcof.py
def fib(n: int) -> int:
    if n == 0:
        return 0
    if n == 1:
        return 1
    
    f1, f2 = 0, 1
    for i in range(n - 1):
        temp = (f1 + f2) % 1000000007
        f1, f2 = f2, temp
    
    return f2
    
if __name__ == "__main__":
    print(fib(5))